k8s/client.go

```diff
@@ -1084,6 +1085,39 @@ func (c *Client) GetHelmState(ctx context.Context, namespace string, secretName
 	}, nil
 }
 
+func (c *Client) GetRESTClientGetter() (*genericclioptions.RESTClientGetter, error) {
+	if c.RESTClientGetter == nil {
+		return nil, fmt.Errorf("unable to get the RESTClientGetter")
+	}
+	return &c.RESTClientGetter, nil
+}
+
+// GetHelmValues is the function for cilium cli sysdump to collect the helm values from the release directly
+func (c *Client) GetHelmValues(_ context.Context, releaseName string, namespace string) (string, error) {
+	restClientGetter, err := c.GetRESTClientGetter()
```

Comment:
Rather than add this function, how about just performing the `nil` check here?

Reply:
My understanding is we have to keep this function since `RESTClientGetter` the interface and I can't just get the struct if I dont implement this function. Isn't it?

Reply:
You're implementing `GetRESTClientGetter` with the receiver `(c *Client)`, so the only interface that you could be satisfying with that implementation would be an interface for `*Client`. However, you are not passing the value of `c` in this function as any interface that expects a `GetRESTClientGetter` method.

Therefore you can replace this line and the function above with:
```
	if c.RESTClientGetter == nil {
		return nil, fmt.Errorf("unable to get RESTClientGetter for Helm Values")
	}
```

And then later in this function use `&c.RESTClientGetter` wherever you are using `restClientGetter`.